.github/workflows/artifact-redirector.yml

```diff
@@ -15,7 +15,7 @@ jobs:
     name: Run CircleCI artifacts redirector
     steps:
       - name: GitHub Action step
-        uses: larsoner/circleci-artifacts-redirector-action@master
+        uses: scientific-python/circleci-artifacts-redirector-action@4e13a10d89177f4bfc8007a7064bdbeda848d8d1  # v1.0.0
```

Comment:
```suggestion
        uses: scientific-python/circleci-artifacts-redirector-action@v1
```

I think our policy is to use major versions instead of hash pinning to limit churn. See the discussion in https://github.com/scikit-learn/scikit-learn/pull/29203#issuecomment-2154372437 (and above and below).

Reply:
I agree with Olivier about using major version, right now my personal simplified summary from the discussion https://github.com/scikit-learn/scikit-learn/pull/29203 and https://github.com/scikit-learn/scikit-learn/pull/29180 is the following:
- do the simplest thing and use major versions for everything, except for some actions where security is crucial, like uploading to PyPI. Clearly not a security expert, but it does not seem like circle-artifacts-redirector-action is not crucial for security so let's stick with major versions (statuses: write for Github, and a CircleCI token not sure about the permissions and what you can do with it)
- wait for SPEC8 to clarify some (hopefully) simple rules to follow and revisit once it has ...
- revisit if we bump into annying things that we did not anticipate